azure-devops/azext_devops/dev/common/github_credential_manager.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import base64
+import requests
+from knack.prompting import prompt, prompt_pass
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.dev.common.utils import randomword, singleton
+
+logger = get_logger(__name__)
+
+
+@singleton
+class GithubCredentialManager():
+    """ GithubCredentialManager
+    """
+    def __init__(self):
+        self.username = None
+        self.password = None
+        self.token = None
+
+    def create_token(self, note=None):
+        self.username = prompt(msg='Enter your GitHub username (leave blank for using already generated PAT): ')
+        if not self.username:
+            self.token = prompt_pass(msg='Enter your GitHub PAT: ')
+            return self.token
+        self.password = prompt_pass(msg='Enter your GitHub password: ')
+        if not note:
+            note = "AzureDevopsCLIExtensionToken_" + randomword(10)
+        encoded_pass = base64.b64encode(self.username.encode('utf-8') + b':' + self.password.encode('utf-8'))
+        basic_auth = 'basic ' + encoded_pass.decode("utf-8")
+        request_body = {
+            "scopes": [
+                'admin:repo_hook',
+                'repo',
+                'user'
+                ],
+            "note": note
+            }
+        headers = {'Content-Type': 'application/json' + '; charset=utf-8',
+                   'Accept': 'application/json',
+                   'Authorization': basic_auth}
+        response = self.post_authorization_request(headers=headers, body=request_body)
+        if (response.status_code == 401 and response.headers.get('X-GitHub-OTP') and
```

Comment:
it will be good to add comments on these logics on where to get this logic from.
I think this is coming from 
https://developer.github.com/v3/auth/